cli/replicaset/configsource.go

```diff
@@ -142,6 +216,63 @@ func (c *CConfigSource) Expel(ctx ExpelCtx) error {
 	)
 }
 
+// AddRole patches a config to add role to a config.
+func (c *CConfigSource) AddRole(ctx RolesAddCtx) error {
+	return c.patchConfigWithRoles(ctx, getCConfigRolesPath, patchCConfigAddRole)
+}
+
+// getCConfigRolesPath returns a path and it's minimum interesting depth
+// to patch the config for role addition.
+func getCConfigRolesPath(clusterConfig libcluster.ClusterConfig, ctx RolesAddCtx) ([]path, error) {
+	var paths []path
+	if ctx.IsGlobal {
+		paths = append(paths, path{
+			path:  []string{"roles"},
+			depth: 0,
+		})
+	}
+	if ctx.GroupName != "" {
+		p := []string{"groups", ctx.GroupName}
+		if _, err := clusterConfig.RawConfig.Get(p); err != nil {
+			var notExistErr libcluster.NotExistError
+			if errors.As(err, &notExistErr) {
+				return []path{}, fmt.Errorf("cannot find group %q", ctx.GroupName)
+			}
+			return []path{}, err
```

Comment:
So why we don't wrap the error here?

Reply:
We wrap error if we know that it was `NotExistError`. In other cases we don't know what error was returned from `Get()` function and that is why we don't wrap error for the other cases.

Reply:
I think we could add some context here to make it easier to detect a code where an error was thrown:
`fmt.Errorf("failed to build a group path: %s", err)`

Reply:
Wrapped.